src/main/java/io/netty/buffer/api/Buf.java

```diff
@@ -320,11 +320,11 @@ default ByteIterator iterate() {
      * @param fromOffset The offset into the buffer where iteration should start.
      *                  The first byte read from the iterator will be the byte at this offset.
      * @param length The number of bytes to iterate.
-     * @return A {@link ByteIterator} for the given stretch of bytes of this buffer.
+     * @return A {@link ByteCursor} for the given stretch of bytes of this buffer.
      * @throws IllegalArgumentException if the length is negative, or if the region given by the {@code fromOffset} and
      * the {@code length} reaches outside of the bounds of this buffer.
      */
-    ByteIterator iterate(int fromOffset, int length);
+    ByteCursor iterate(int fromOffset, int length);
```

Comment:
should we change the method name to `cursor` ?

Reply:
Same comment for the other methods.

Reply:
Considered it, but `cursor` is not a verb. I've used `open` or `openCursor` previously. How does that sound?

Reply:
`openCursor` sounds good imho